pkg/agent/monitoraddnodes.go

```diff
@@ -68,16 +69,34 @@ func (mon *addNodeMonitor) logStatus(status string) {
 	logrus.Infof("Node %s: %s", mon.nodeIPAddress, status)
 }
 
-// MonitorAddNodes waits for the a node to be added to the cluster
+// MonitorAddNodes display the progress of one or more nodes being
+// added to a cluster. ipAddresses is an array of IP addresses to be
+// monitored. clusters is an array of their corresponding initialized Cluster
+// struct used to interact with the assisted-service and k8s APIs.
+func MonitorAddNodes(clusters []*Cluster, ipAddresses []string) {
+	var wg sync.WaitGroup
+
+	for i, ipAddress := range ipAddresses {
+		wg.Add(1)
+
+		go MonitorSingleNode(clusters[i], ipAddress, &wg)
+	}
+
+	wg.Wait()
+}
+
+// MonitorSingleNode waits for the a node to be added to the cluster
 // and reports its status until it becomes Ready.
-func MonitorAddNodes(cluster *Cluster, nodeIPAddress string) error {
+func MonitorSingleNode(cluster *Cluster, nodeIPAddress string, wg *sync.WaitGroup) {
+	defer wg.Done()
 	timeout := 90 * time.Minute
 	waitContext, cancel := context.WithTimeout(cluster.Ctx, timeout)
 	defer cancel()
 
 	mon, err := newAddNodeMonitor(nodeIPAddress, cluster)
 	if err != nil {
-		return err
+		logrus.Errorf("could not initialize node monitor for node %v: %v", nodeIPAddress, err)
```

Comment:
rather than simply logging an error condition in the gorouting, it could be useful to setup a channel for the error handling. In that way it would be possible to collect all the errors outside each goroutine in a central location. This could help in producing a single consolidated output at the end of the monitor operations.

Reply:
Maybe this should be differed. Currently the logging is in real time. It is unclear to me how the single consolidated output would be used.

Reply:
Let's say a single error channel is created in the ` MonitorAddNodes`, and passed as a parameter to each goroutine for `MonitorSingleNode`, ie something like that: 

```
errCh := make(chan error)
...
go MonitorSingleNode(waitContext, clusters[i], ipAddress, &wg, errCh)
```

`MonitorSingleNode` sends (any) error detected in the channel, ie: 
```
	mon, err := newAddNodeMonitor(nodeIPAddress, cluster)
	if err != nil {
             errCh <- fmt.Errorf("could not initialize node monitor for node %v: %v", nodeIPAddress, err)
        }
```

There could be then a separate gorouting for receiving all the errors and store them in a slice, ie: 

```
...
go func() {
  for err := range errCh {
      logrus.Error(err)
      allErrs = append(allErrs, err)
  }
}
```

This approach IMHO could allow probably a better logging, and moreover collecting the errs will allow to report them in json struct at the end of the command.

Note that this is not a blocking point for me, I'm fine to proceed with the current (error handling) implementation and review it/improve subsequently if needed